packages/@aws-cdk/aws-apprunner-alpha/lib/observability-configuration.ts

```diff
@@ -0,0 +1,169 @@
+   */
+  public static fromObservabilityConfigurationAttributes(scope: Construct, id: string,
+    attrs: ObservabilityConfigurationAttributes): IObservabilityConfiguration {
+    const observabilityConfigurationArn = attrs.observabilityConfigurationArn;
+    const observabilityConfigurationName = attrs.observabilityConfigurationName;
+    const observabilityConfigurationRevision = attrs.observabilityConfigurationRevision;
+
+    class Import extends cdk.Resource {
+      public readonly observabilityConfigurationArn = observabilityConfigurationArn
+      public readonly observabilityConfigurationName = observabilityConfigurationName
+      public readonly observabilityConfigurationRevision = observabilityConfigurationRevision
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Observability Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, observabilityConfigurationArn: string): IObservabilityConfiguration {
+    const arn = cdk.Stack.of(scope).splitArn(observabilityConfigurationArn, cdk.ArnFormat.SLASH_RESOURCE_NAME);
+
+    const resourceParts = arn.resourceName?.split('/');
+
+    if (!resourceParts || resourceParts.length < 3) {
+      throw new Error(`Unexpected ARN format: ${observabilityConfigurationArn}`);
+    }
+
+    const observabilityConfigurationName = resourceParts[0];
+    const observabilityConfigurationRevision = parseInt(resourceParts[1]);
+
+    return ObservabilityConfiguration.fromObservabilityConfigurationAttributes(scope, id, {
+      observabilityConfigurationArn,
+      observabilityConfigurationName,
+      observabilityConfigurationRevision,
+    });
+  }
+
+  /**
+   * The ARN of the Observability configuration.
+   * @attribute
+   */
+  readonly observabilityConfigurationArn: string;
+
+  /**
+   * The revision of the Observability configuration.
+   * @attribute
+   */
+  readonly observabilityConfigurationRevision: number;
+
+  /**
+   * The name of the Observability configuration.
+   * @attribute
+   */
+  readonly observabilityConfigurationName: string;
+
+  public constructor(scope: Construct, id: string, props: ObservabilityConfigurationProps = {}) {
+    super(scope, id, {
+      physicalName: props.observabilityConfigurationName,
+    });
+
+    if (props.observabilityConfigurationName !== undefined) {
+      if (props.observabilityConfigurationName.length < 4 || props.observabilityConfigurationName.length > 32) {
+        throw new Error(`observabilityConfigurationName must be between 4 and 32 characters long, but it has ${props.observabilityConfigurationName.length} characters.`);
+      }
+
+      if (!/^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$/.test(props.observabilityConfigurationName)) {
+        throw new Error(`observabilityConfigurationName ${props.observabilityConfigurationName} must start with a letter or number, and can contain only letters, numbers, hyphens, and underscores.`);
+      }
+    }
```

Comment:
```suggestion
    if (
      props.observabilityConfigurationName !== undefined &&
      !cdk.Token.isUnresolved(props.observabilityConfigurationName) &&
      !/^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$/.test(props.observabilityConfigurationName)
    ) {
      throw new Error(`observabilityConfigurationName must match the \`^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$\` pattern, got ${props.observabilityConfigurationName}`);
    }
```
Similar comment to the auto scaling validation PR